File: FacebookButttologinNew/app/src/main/java/com/example/facebookbutttologinnew/MainActivity.java
package com.example.facebookbutttologinnew;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.facebook.CallbackManager;
import com.facebook.FacebookCallback;
import com.facebook.FacebookException;
import com.facebook.FacebookSdk;
import com.facebook.appevents.AppEventsLogger;
import com.facebook.login.LoginManager;
import com.facebook.login.LoginResult;

import java.util.Arrays;

public class MainActivity extends AppCompatActivity {
Button fbbtn;
CallbackManager callbackManager;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        callbackManager = CallbackManager.Factory.create();
        LoginManager.getInstance().registerCallback(callbackManager,
                new FacebookCallback<LoginResult>() {
                    @Override
                    public void onSuccess(LoginResult loginResult) {
                        // App code
                        Intent intent =new Intent(MainActivity.this,SecondActivity.class);
                        startActivity(intent);
                    }

                    @Override
                    public void onCancel() {
                        // App code
                        Toast.makeText(getApplicationContext(),"User canceled login",Toast.LENGTH_LONG).show();
                    }

                    @Override
                    public void onError(FacebookException exception) {
                        // App code
                        Toast.makeText(getApplicationContext(),"error => "+exception.toString(),Toast.LENGTH_LONG).show();
                    }
                });
        fbbtn = findViewById(R.id.facebookbtn);
        fbbtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {


                LoginManager.getInstance().logInWithReadPermissions(MainActivity.this, Arrays.asList("public_profile"));
            }
        });
    }


    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        callbackManager.onActivityResult(requestCode, resultCode, data);
        super.onActivityResult(requestCode, resultCode, data);
    }
}

File: SharePreferencekotlin/app/src/main/java/com/example/sharepreferencekotlin/MainActivity.kt
package com.example.sharepreferencekotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView


class MainActivity : AppCompatActivity() {
   lateinit var insertbtn: Button
   lateinit var showbtn: Button
   lateinit var deleltbtn: Button
   lateinit var movetorecyclerview: Button
   lateinit var showtext: TextView
   lateinit var editvalue: EditText
   lateinit var editkey: EditText
    lateinit var key: String
   lateinit var sharedoperation :SharedOperations
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        conecxml()
        sharedoperation = SharedOperations()
        insertbtn.setOnClickListener {
            key = editkey.text.toString()
            val value = editvalue.text.toString()
            sharedoperation.insertvalue(key, value, applicationContext)
        }
        showbtn.setOnClickListener {
            key = editkey.text.toString()
            val vlaue: String = sharedoperation.showvalue(key, applicationContext)!!
            showtext.text = vlaue
        }
        deleltbtn.setOnClickListener {
            key = editkey.text.toString()
            sharedoperation.deletekey(key, applicationContext)
        }
        movetorecyclerview.setOnClickListener {
            startActivity(
                Intent(
                    this@MainActivity,
                    SecondActivity::class.java
                )
            )
        }
    }

    private fun conecxml() {
        insertbtn = findViewById(R.id.insertbtn)
        showbtn = findViewById(R.id.Showbtn)
        showtext = findViewById(R.id.valuetextview)
        editvalue = findViewById(R.id.value)
        editkey = findViewById(R.id.keyedit)
        deleltbtn = findViewById(R.id.deletebtn)
        movetorecyclerview = findViewById(R.id.movetoanotherbtn)
    }
}